src/pyk/kcfg/kcfg.py

```diff
@@ -968,6 +969,24 @@ def prune(self, node_id: str) -> None:
         for node in nodes:
             self.remove_node(node.id)
 
+    def shortest_path_between(self, source_node_id: str, target_node_id: str) -> tuple[Successor, ...] | None:
+        paths = self.paths_between(source_node_id, target_node_id)
+        return sorted(paths, key=(lambda path: len(path)))[0]
+
+    def path_constraints(self, final_node_id: str) -> KInner:
+        path = self.shortest_path_between(self.get_unique_init().id, final_node_id)
+        if path is None:
+            raise ValueError(f'No path found to specified node: {final_node_id}')
+        curr_constraint: KInner = mlTop()
+        for edge in reversed(path):
+            if type(edge) is KCFG.Split:
+                assert len(edge.targets) == 1
+                _, csubst = edge.targets[0]
+                curr_constraint = mlAnd([csubst.constraint, curr_constraint])
+            if type(edge) is KCFG.Cover:
+                curr_constraint = edge.csubst.apply(CTerm.from_kast(curr_constraint)).kast
```

Comment:
```suggestion
                curr_constraint = mlAnd([edge.csubst.constraint, edge.csubst.subst.apply(curr_constraint)])
```

Turning `curr_constraint` into a `CTerm` will probably _always_ fail, because it does not have a configuration.

Instead, we should project out the `subst` and just apply that to the constraint directly as a `KInner`.

We could also add the `edge.csubst.constraint` with the above `mlAnd` added too. That should be _ok_ to do, because that constarint should already exist on the path leading up to this point for us to add it like that. We may just not want to include those constraints eventually, but we can figure that out later.

Reply:
I changed this.